l2geth/rollup/sync_service.go

```diff
@@ -162,10 +175,33 @@ func NewSyncService(ctx context.Context, cfg Config, txpool *core.TxPool, bc *co
 			service.setSyncStatus(true)
 		}
 	}
-
 	return &service, nil
 }
 
+func (s *SyncService) transition() {
+	for {
+		select {
+		case tx := <-s.txCh:
```

Comment:
It is important that this channel is only big enough for 1 thing otherwise reorgs will be a pain since there will be txs buffered in the channel and we would have to empty the channel. If done like that then the lock is a bit redundant